.github/workflows/cli-tests.yml

```diff
@@ -0,0 +1,213 @@
+                'daemon-controller': 'winsvc',
+                'sudo-cmd': 'gsudo',
+                'pkg-url': WINDOWS_PKG_URL,
+                'pkg-download-destination': '$env:RUNNER_TEMP\\multipass.msi',
+                'cmd-prologue': '',
+                'concurrency-key': shortHash(WINDOWS_PKG_URL)
+              });
+            }
+
+            core.setOutput('matrix', JSON.stringify(matrix));
+            core.setOutput('has_targets', String(matrix.include.length > 0));
+  run-cli-tests:
+    concurrency:
+      group: cli-tests-${{ github.workflow }}-${{ github.head_ref || github.run_id }}-${{ matrix.concurrency-key }}
+      cancel-in-progress: true
+    name: CLI Tests (${{ matrix.label }})
+    needs: determine-matrix
+    runs-on: ${{ matrix.os }}
+    if: ${{ needs.determine-matrix.outputs.has_targets == 'true' }}
+    strategy:
+      fail-fast: false
+      matrix: ${{ fromJSON(needs.determine-matrix.outputs.matrix) }}
+    steps:
+      - name: Check out cli-tests
+        uses: actions/checkout@v5
+        with:
+          fetch-depth: 1
+          submodules: false
+          sparse-checkout: |
+            tools/cli_tests
+
+      - name: Download package with cURL
+        if: ${{ matrix['pkg-url'] != '' }}
+        run: curl -fL "${{ matrix['pkg-url'] }}" -o ${{ matrix['pkg-download-destination'] }}
+
+      ##################################################################
+      # Platform-specific
+      ##################################################################
+      - name: Install prerequisites (Windows)
+        if: runner.os == 'Windows'
+        run: |
+          choco install gsudo -y
+          echo 'C:\ProgramData\chocolatey\bin' >> "$GITHUB_PATH"
+          & "C:\tools\gsudo\Current\gsudo.exe" -v
+          echo "C:\tools\gsudo\Current" | Out-File -FilePath $env:GITHUB_PATH -Encoding utf8 -Append
+
+      - name: Install package (Snap)
+        if: runner.os == 'Linux'
+        run: ${{ matrix.cmd-prologue }}${{ matrix.sudo-cmd }} snap install multipass --channel=${{ matrix['snap-channel'] }}
+
+      - name: Install package (macOS)
+        if: runner.os == 'macOS'
+        run: ${{ matrix.cmd-prologue }} ${{ matrix.sudo-cmd }} installer -pkg ${{ matrix['pkg-download-destination'] }} -target /
+
+      - name: Install package (Windows)
+        if: runner.os == 'Windows'
+        run: |
+          $msi = "${{ matrix['pkg-download-destination'] }}"
+          $log = Join-Path $env:RUNNER_TEMP 'multipass-install.log'
+          Write-Host "Expanded: $msi"
+
+          $args = @(
+            '/i', $msi,
+            '/qn', '/norestart',
+            '/l*vx!', $log,
+            'ADDLOCAL=ALL',
+            'ALLUSERS=1'
+          )
+
+          $proc = Start-Process -FilePath msiexec.exe -ArgumentList $args -Wait -PassThru
+          $code = $proc.ExitCode
+
+          Write-Host "msiexec exit code: $code"
+          if (Test-Path $log) {
+            Get-Content -Path $log -Tail 200
```

Comment:
Why not print out all the logs? This could indirectly allow us to test the installer.

Reply:
Logs from compilation are fairly extensive as well and they can always be downloaded into a text file for easier inspection. Again, I would just tend to air on the side of collecting more information, not less. Its also just debug info, so hopefully we don't have to look at it too much.